updated Dashboard, Pivot EUD 20.1 (#107)
DevExpress/dotnet-eud: dashboard-for-web/articles/web-dashboard-designer-mode/appearance-customization/conditional-formatting.md

legacyId: 117564
---
# Conditional Formatting
The Web Dashboard supports conditional formatting. You can apply a custom style to data elements that satisfy a certain condition. 

> For more information about conditional formatting specifics for different dashboard items, see the following topics:

Reviewer: ```suggestion
For more information about conditional formatting for different dashboard items, see the following topics:
```

* [Conditional Formatting - Grid](../designing-dashboard-items/grid/conditional-formatting.md)
* [Conditional Formatting - Pivot](../designing-dashboard-items/pivot/conditional-formatting.md)
* [Conditional Formatting - Card](../designing-dashboard-items/cards/conditional-formatting.md)

![wdd-cf-main](../../../images/img126130.png)

> Format rules used in conditional formatting can be divided into the following groups:

Reviewer: ```suggestion
Format rules used in conditional formatting can be categorized by the following groups:
```

> * **Value** - Allows you to compare static values (such as Greater Than, Less Than, Between, etc.).

Reviewer: ```suggestion
* **Value** - Compares static values (Greater Than, Less Than, Between, etc.).
```

> * **Top-Bottom** - Highlights a specific number of topmost/bottommost values.

Reviewer: ```suggestion
* **Top-Bottom** - Highlights a specific number of top/bottom values (Top N, Bottom N)
```

> * **Average** - Highlights values above the average value or below the average value.

Reviewer: ```suggestion
* **Average** - Highlights values above or below the average value.
```

> * **A Date Occurring** - Allows you to highlight date-time values that fall into a specified interval.

Reviewer: ```suggestion
* **A Date Occurring** - Highlights date-time values that are within a specified interval.
```

> * **Expression** - Allows you to use complex conditions to apply formatting. You can also pass dashboard parameters to expressions.

Reviewer: ```suggestion
* **Expression** - Uses complex conditions to apply formatting. You can also pass dashboard parameters to expressions.
```

> * **Icon and Color Ranges** - Allows you to apply formatting by displaying specific icons for different ranges of values. You can select a predefined set of icons or use a specific icon for each range.

Reviewer: ```suggestion
* **Icon and Color Ranges** - Displays specific icons for different value ranges to apply formatting. You can select a predefined set of icons or apply a specific icon to each range.
```

> * **Color Ranges** - Allows you to apply formatting using specific colors for different ranges of values. You can select a predefined set of colors or use custom appearance settings to highlight values within specified ranges.

Reviewer: ```suggestion
* **Color Ranges** - Apply specific colors to different value ranges. You can select a predefined set of colors or use custom appearance settings to highlight values within specific ranges.
```

> * **Bar Color Ranges** - Allows you to visualize numeric values using bars whose colors are contained in the specified color set.

Reviewer: I'd like to clarify this definition. My understanding is that each color represents a range of values. If the value falls within a specific range, the color specified by that range is displayed. Is this correct?
If so, can we say:

Bar Color Ranges - Visualizes numeric values as bars. Values within a range display a specific color.

Author: ```suggestion
* **Bar Color Ranges** - Visualizes numeric values as bars. Values within a range display a specific color.
```

Reviewer: Please change to:

* **Bar Color Ranges** - Visualize numeric values as bars. Values within a range display a specific color.

> * **Bar Gradient Ranges** - Allows you to visualize numeric values using bars whose colors are contained in the specified color gradient.

Reviewer: Please see my similar comment above. If my understanding is correct, would it be technically correct to say:

Bar Gradient Ranges - Visualizes numeric values as bars. Values within a range display a specific bar gradient.

Author: ```suggestion
* **Bar Gradient Ranges** - Visualizes numeric values as bars. Values within a range display a specific color.
```

Reviewer: Change to:

* **Bar Gradient Ranges** - Visualize numeric values as bars. Values within a range display a specific color.

> You can calculate a format rule using [measure or dimension](../binding-dashboard-items-to-data/binding-dashboard-items-to-data-in-the-web-dashboard.md) values. For the Card dashboard item, format rules also can be calculated by [delta](../designing-dashboard-items/cards/delta.md) values.

Reviewer: ```suggestion
You can use [measure or dimension](../binding-dashboard-items-to-data/binding-dashboard-items-to-data-in-the-web-dashboard.md) values to calculate a format rule. For a Card dashboard item, you can also calculate format rules by [delta](../designing-dashboard-items/cards/delta.md) values.
```

The table below lists format conditions that can be applied to different types of data items.
* numeric 
	* **Value**
	* **Top-Bottom**
	* **Average**
	* **Expression**
	* **Icon Ranges**
	* **Color Ranges**
	* **Gradient Ranges**
	* **Bar** <sup>1</sup>
	* **Bar Color Ranges** <sup>1</sup>
	* **Bar Gradient Ranges** <sup>1</sup>
* string 

> 	* **Value** with the condition type set to _Equal To_, _Not Equal To_ or _Text that Contains_

Reviewer: ```suggestion
	* **Value** (with the condition type set to _Equal To_, _Not Equal To_ or _Text that Contains_)
```

	* **Expression**
* date-time 
	* **Value**

> 	* A **Date Occurring** for dimensions with the continuous date-time group interval

Reviewer: ```suggestion
	* A **Date Occurring** (for dimensions with a continuous date-time group interval)
```

	* **Bar Gradient Ranges** <sup>1</sup>

	<sup>1</sup>  The Card dashboard item does not support these rules.

## Create a Format Rule

 To create a format rule, open the **Conditional Formatting** section in the dashboard item's [Options](../ui-elements/dashboard-item-menu.md) menu or in the [data item menu](../ui-elements/data-item-menu.md). Click "+" to add a new format rule:
	
![wed-dashboard-cf-add-rule](../../../images/wed-dashboard-cf-add-rule.png)

Specify the data item/card used to calculate a condition in the **Common** section. You can also create a format rule for one data item and apply different settings to the other data item. 

Select a format rule type from the list to open its settings.
	
![wdd-grid-cf-select-rule-type](../../../images/img126024.png)

Select a condition from the list and [specify its settings](#appearance-settings) in the **Condition** section. Available settings depend on the selected format rule.
	
![wdd-grid-cf-value-menu](../../../images/img126023.png)
	
Specify additional settings in the **Miscellaneous** section. For example, you can specify the intersection level for the Pivot or apply the current rule to a row in the Grid.

> See the following topics for more information about specific formatting settings of the dashboard items:

Reviewer: ```suggestion
See the following topics for more information about specific format settings for dashboard items:
```

* [Conditional Formatting - Grid](../designing-dashboard-items/grid/conditional-formatting.md)
* [Conditional Formatting - Pivot](../designing-dashboard-items/pivot/conditional-formatting.md)
* [Conditional Formatting - Card](../designing-dashboard-items/cards/conditional-formatting.md)

## Edit a Format Rule

> You can see existing format rules for the entire dashboard item. To do this, open the dashboard item's [Options](../ui-elements/dashboard-item-menu.md) menu and go to the **Conditional Formatting** section.

Reviewer: ```suggestion
You can see existing format rules for an entire dashboard item. To do this, open the dashboard item's [Options](../ui-elements/dashboard-item-menu.md) menu and go to the **Conditional Formatting** section.
```

![wdd-cf-all-rules](../../../images/img126046.png)

To edit a format rule, select the rule and click **Edit** (the ![wdd-icon-edit-collection-value-item](../../../images/img126050.png) icon).

Click **Delete** (the ![wdd-icon-delete-big](../../../images/img126104.png) icon) to delete the selected format rule.

When you edit a format rule, you can enable or disable the rule in the **Miscellaneous** section.

![](../../../images/web-conditional-formatting-edit-rule-miscellaneous-section.png)

## Appearance Settings

> The format rule's menu **Condition** section contains appearance settings. You can configure and customize current format condition appearance settings:

Reviewer: ```suggestion
The format rule menu's **Condition** section contains appearance settings. You can configure and customize current format condition appearance settings:
```
